matrixprofile/utils.py

```diff
@@ -12,6 +12,272 @@
 from matrixprofile import core
 
 
+def apply_av(profile, av="default", custom_av=None):
```

Comment:
This function has a problem with mutating the data structure before all of the validation passes. Here is a silly example you can play with to see the issue:

```python
profile = {'key': 1}

def change(p):
    p['key'] = 2

change(profile)
print(profile)
```
Note that in this silly example the key is changed similar to how you set the av and av_cmp keys. A better approach is to set these keys to variables and at the end mutate the profile data structure.